### PR TITLE
Wait for graceful MQTT disconnect

### PR DESCRIPTION
**Pull request by feymartynov**
On destroy MQTT transport publishes a offline status message and then calls `MQTTAsync_disconnect` which gracefully disconnects from the broker by sending `disconnect` message. However this call is asynchronous and is not being awaited and the socket closes faster than the message is being sent. From the point of view of the broker it looks like ungraceful disconnection so it publishes another one offline status message as LWT. As a result subscribers get duplicated offline status messages.

This PR fixes this problem by synchronizing `MQTTAsync_disconnect` call using `janus_condition_wait_until`.

### CODE REVIEW

File: transports/janus_mqtt.c [hunk 1]
```diff
@@ -1224,7 +1228,16 @@ int janus_mqtt_client_disconnect(janus_mqtt_context *ctx) {
 
 	options.context = ctx;
 	options.timeout = ctx->disconnect.timeout;
-	return MQTTAsync_disconnect(ctx->client, &options);
+	int rc = MQTTAsync_disconnect(ctx->client, &options);
+	if (rc == MQTTASYNC_SUCCESS) {
+		janus_mutex_lock(&ctx->disconnect.mutex);
+		// Block the thread awaiting asynchronous graceful disconnect to finish.
```
Comment (lminiero) [inline transports/janus_mqtt.c:1234]: Code style: comments should be in `/* */` format.

**Review by lminiero — COMMENTED** [review]
I assume you tested this, and that it now works as expected, so apart from the note I added this looks good to me. As soon as the note is addressed I'll merge, thanks :+1:

File: transports/janus_mqtt.c (hunk 1 above)
Comment (feymartynov) [reply transports/janus_mqtt.c]: Fixed

**Comment by feymartynov** [comment]
Sure, I've tested it.

**Comment by lminiero** [comment]
Thanks, merging then :+1:

**Pull request merged**